docs/src/index.md

```diff
@@ -315,3 +315,55 @@ To get back an `Array` from a `DArray`, just call `collect`:
 DA = rand(Blocks(32, 32), 256, 128)
 collect(DA) # returns a `Matrix{Float64}`
 ```
+
+## Quickstart: Datadeps
+
+Datadeps is a feature in Dagger.jl that facilitates parallelism control within designated regions, allowing tasks to write to their arguments while ensuring dependencies are respected.
+For more details: [Datadeps](@ref)
+
+### Syntax
+The Dagger.spawn_datadeps() function is used to create a "datadeps region" where tasks are executed with parallelism controlled by specified dependencies.
+
+```julia
+Dagger.spawn_datadeps() do
+    Dagger.@spawn func!(Out(var_name_x), In(var_name_y))
+end
+```
+### Argument Annotation
+
+- In(X): Indicates that the variable X is an input to the task.
+- Out(X): Indicates that the variable X is an output of the task.
+- InOut: Indicates that the variable is both an input and an output.
```

Comment:
```suggestion
- `In(X)`: Indicates that the variable `X` is only read by the task (an "input").
- `Out(X)`: Indicates that the variable `X` is only written to by the task (an "output").
- `InOut(X)`: Indicates that the variable `X` is both read from and written to by the task (an "input" and "output" simultaneously).
```

To indicate what "input" and "output" really signify as far as task behavior is concerned.